src/libs/HttpUtils.js

```diff
@@ -13,9 +13,16 @@ Onyx.connect({
 });
 
 let shouldFailAllRequests = false;
+let shouldForceOffline = false;
 Onyx.connect({
     key: ONYXKEYS.NETWORK,
-    callback: val => shouldFailAllRequests = (val && _.isBoolean(val.shouldFailAllRequests)) ? val.shouldFailAllRequests : false,
+    callback: (network) => {
+        if (!network) {
+            return;
+        }
+        shouldFailAllRequests = _.isBoolean(network.shouldFailAllRequests) ? network.shouldFailAllRequests : false;
+        shouldForceOffline = Boolean(network.shouldForceOffline);
```

Comment:
Yeah good point I like that better. I'm wondering whether it's better to use the Boolean constructor in general or something like `definitelyBoolean = network.maybeBoolean || false`. I think the constructor is probably preferred since it will work better in a large expression like
`offline = network.isOffline || Boolean(network.shouldForceOffline);`
vs
`offline = network.isOffline || (network.shouldForceOffline || false);`